test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs

```diff
@@ -744,24 +746,23 @@ await SimpleTestPackageUtility.CreatePackagesAsync(
             mockServer.Start();
 
             _fixture.RunDotnetExpectSuccess(Directory.GetParent(projectA.ProjectPath).FullName, $"add package A --version 1.0.0", testOutputHelper: _testOutputHelper);
+            PackageSource packageSource = new PackageSource(mockServer.ServiceIndexUri, "http-source");
+            pathContext.Settings.AddSource(packageSource.Name, packageSource.Source, allowInsecureConnections);
 
-            pathContext.Settings.AddSource("http-source", mockServer.ServiceIndexUri, allowInsecureConnections);
-
-            // Act
-            CommandRunnerResult listResult = _fixture.RunDotnetExpectSuccess(Directory.GetParent(projectA.ProjectPath).FullName, $"list package --outdated --no-restore", testOutputHelper: _testOutputHelper);
-            mockServer.Stop();
-
-            // Assert
-            var lines = listResult.AllOutput.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
-            Assert.True(lines.Any(l => l.Contains("> A                    1.0.0       1.0.0      2.0.0")), listResult.AllOutput);
-            if (isHttpWarningExpected)
+            // Act & Assert
+            if (isHttpErrorExpected)
```

Comment:
Maybe in the future, we can consider refactoring this test to remove the conditional. Typically, this might mean splitting this into multiple tests and reusing the common setup code to avoid duplication.